### PR TITLE
Updated Edge Core to v0.15.0, added options to compile new FOTA lib

### PR DESCRIPTION
**Pull request by ygoyal18**
1. Updated EC to 0.15.0.
1. Added patch to fix mbed-cloud-client CMake. Will have to remove it once PDMC v4.7.1 is released with new version of EC.
1. Added options to compile EC with new FOTA framework library.
1. Added option to either statically or dynamically link libcurl. Static linking is not working, it is advised to enable dynamic link when using new fota library.

### CODE REVIEW

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc [hunk 1]
```diff
@@ -56,6 +59,8 @@ EXTRA_OECMAKE += " \
     -DTARGET_CONFIG_ROOT=${WORKDIR} \
     -DTRACE_LEVEL=${MBED_EDGE_CORE_CONFIG_TRACE_LEVEL} \
     -DFIRMWARE_UPDATE=${MBED_EDGE_CORE_CONFIG_FIRMWARE_UPDATE} \
+    -DMBED_CLOUD_CLIENT_CURL_DYNAMIC_LINK=${MBED_EDGE_CORE_CONFIG_CURL_DYNAMIC_LINK} \
+    -DFOTA_ENABLE=${MBED_EDGE_CORE_CONFIG_FOTA_ENABLE} \
     -DDEVELOPER_MODE=${MBED_EDGE_CORE_CONFIG_DEVELOPER_MODE} \
```
Comment (petedyerarm) [inline recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:64]: -DFOTA_TRACE=${MBED_EDGE_CORE_CONFIG_FOTA_TRACE} \

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc [hunk 2]
```diff
@@ -5,14 +5,16 @@ LIC_FILES_CHKSUM = "file://${WORKDIR}/git/LICENSE;md5=1dece7821bf3fd70fe1309eaa3
 
 MBED_EDGE_CORE_CONFIG_TRACE_LEVEL ?= "DEBUG"
 MBED_EDGE_CORE_CONFIG_FIRMWARE_UPDATE ?= "OFF"
+MBED_EDGE_CORE_CONFIG_FOTA_ENABLE ?= "OFF"
```
Comment (petedyerarm) [inline recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:8]: Perhaps add MBED_EDGE_CORE_CONFIG_FOTA_TRACE ?= "OFF"

**Review by petedyerarm — COMMENTED** [review]
LGTM, but I also have added FOTA_TRACE. 
It is a suggestion that it is added, but not essential.

**Comment by ygoyal18** [comment]
I have added "do not merge" label as @x85446 and I have decided not to allow RPi users to enable new FOTA library as we are not creating the new update scripts and not using osTree to manage the updates in 2.2 release. I will raise another PR which  will upgrade the edge-core version and only allow users to use UC Hub library (enabled by default with FIRMWARE_UPDATE flag) for the firmware update. Will merge this in 2.3 once we have osTree in the RPi builds and we create new update scripts to support that.

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc (hunk 1 above)
Comment (ygoyal18) [reply recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:64]: @petedyerarm Sure, will add that.

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc (hunk 2 above)
Comment (ygoyal18) [reply recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:8]: @petedyerarm Sure, will add that.

**Comment by ygoyal18** [comment]
My bad, for some reason I thought we are compiling the code in bitbake file and not in include file. @x85446 @JanneKiiskila How do we make sure people using RPi do not enable the FOTA flag as we do not have the update scripts to support it.

**Comment by JanneKiiskila** [comment]
Do we have to? Should not be enough to just state it as a known issue in the release notes? Or issue a warning if target == RPI && update== FOTA?

**Comment by ygoyal18** [comment]
Ya lets check for that. @JanneKiiskila are we support UC Hub on imx8? Should we also add a warning for platform == imx8 and FOTA_ENABLE == OFF and FIRMWARE_UPDATE == ON?

**Comment by ygoyal18** [comment]
Merging this. Let's bring in improvements through subsequent PRs.

**Pull request merged**